components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/ServiceBusInitializer.java

```diff
@@ -522,4 +539,17 @@ public void setConfigAdminService(ConfigurationAdmin configAdminService) {
     public void unsetConfigAdminService(ConfigurationAdmin configAdminService) {
         log.debug("Unsetting ConfigurationAdmin Service");
     }
+
+    @Reference(name = "org.wso2.carbon.ndatasource",
```

Comment:
This cannot have an impact on the startup time since we already have a dependency to TaskService which is already dependecnt on the ndatasource component. But shall we double check the startup time, just to be safe?

Reply:
Yes true , in that case do we need to specifically add this as a reference. Shall we neglect it as it is already implied.

Reply:
Checked. I could not notice a time difference in the startup time with this feature enabled.